spring-rabbit/src/test/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactoryIntegrationTests.java

```diff
@@ -255,14 +256,16 @@ public void testSendAndReceiveFromVolatileQueue() throws Exception {
 
 	@Test
 	public void testReceiveFromNonExistentVirtualHost() throws Exception {
-
 		connectionFactory.setVirtualHost("non-existent");
 		RabbitTemplate template = new RabbitTemplate(connectionFactory);
 		// Wrong vhost is very unfriendly to client - the exception has no clue (just an EOF)
-		exception.expect(AmqpIOException.class);
-		String result = (String) template.receiveAndConvert("foo");
-		assertEquals("message", result);
-
+		try {
+			template.receiveAndConvert("foo");
+			fail("Exception expected");
+		}
+		catch (AmqpIOException | AmqpAuthenticationException e) {
+			// expected
```

Comment:
Well, if there is no assertion, we fail with just a raw exception which isn't caught by this `catch` block.
With the test report it isn't clear what's wrong. Otherwise, with an assertion any our future changes make us to think twice to do them, since we have particular test for an appropriate expectation.